### PR TITLE
Move ray.put, ray.get, ray.wait definitions to api.py

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Moves all top-level APIs from `worker.py` to `api.py` except for `ray.init`. The `ray.init` change will be a little more involved to do cleanly, so I'll address that in a follow-up.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
Actually @ericl there are more low-hanging fruit, let me get them all at once to avoid unnecessary merge conflicts, messy reverts, etc.

**Comment by edoakes** [comment]
Ok, PTAL. Basically everything is a copy-paste aside from changing a few import paths and calling into `ray.worker` where necessary.

File: python/ray/__init__.py [hunk 1]
```diff
@@ -73,14 +73,17 @@
...
+    get_dashboard_url, is_initialized, kill, shutdown, show_in_dashboard,
+    remote,
 )
 from ray.worker import (  # noqa: E402,F401
     LOCAL_MODE, SCRIPT_MODE, WORKER_MODE, RESTORE_WORKER_MODE,
-    SPILL_WORKER_MODE, cancel, get, get_actor, get_gpu_ids, get_resource_ids,
-    get_dashboard_url, init, is_initialized, put, kill, remote, shutdown,
-    show_in_dashboard, wait,
-)
+    SPILL_WORKER_MODE, init,
```
Comment (rkooo567) [inline python/ray/__init__.py:84]: Why is init from ray.worker?

**Review by rkooo567 — APPROVED** [review]

File: python/ray/__init__.py (hunk 1 above)
Comment (edoakes) [reply python/ray/__init__.py:84]: See description

**Comment by edoakes** [comment]
Alas, there are now too many conflicts here. Better to start from scratch...

**Pull request closed without merging**